services/QuillLMS/app/workers/batch_save_activity_session_concept_results_worker.rb

```diff
@@ -15,13 +15,27 @@ def perform(concept_results, activity_session_id, activity_session_uid)
   end
 
   def on_complete(status, options)
+    activity_session_uid = options['activity_session_uid']
+
+    delete_student_score_cache(activity_session_uid)
+
     if status.failures == 0
-      PusherTrigger.run(options['activity_session_uid'], 'concept-results-saved', "Concept results saved for activity session with uid: #{options['activity_session_uid']}")
+      PusherTrigger.run(activity_session_uid, 'concept-results-saved', "Concept results saved for activity session with uid: #{options['activity_session_uid']}")
     else
-      PusherTrigger.run(options['activity_session_uid'], 'concept-results-partially-saved', "Concept results partially saved for activity session with uid: #{options['activity_session_uid']}, #{status.total - status.failures} succeeded, #{status.failures} failed")
+      PusherTrigger.run(activity_session_uid, 'concept-results-partially-saved', "Concept results partially saved for activity session with uid: #{options['activity_session_uid']}, #{status.total - status.failures} succeeded, #{status.failures} failed")
     end
   end
 
+  private def delete_student_score_cache(activity_session_uid)
+    session = ActivitySession.find_by(uid: activity_session_uid)
+
+    return unless session
+
+    cache_key = "#{Student::EXACT_SCORES_CACHE_KEY}/#{session.user_id}/#{session.activity_id}/#{session.classroom_unit_id}"
```

Comment:
Looks like we're generating this cache key in two separate places.  Given the limited scope here, that's probably fine, but I did want to flag that in general we've tried to put the key generation logic in shared functions.

Reply:
That's fair, will look at better ways to do this after deploying this (since it's somewhat time sensitive).